R/timeseries_plots.R

```diff
@@ -1,57 +1,74 @@
-  data <- get_var(file, z_out = z_out, var_name = var_name, reference = reference)
-  title = .unit_label(file, var_name)
-  .plot_df_heatmap(data, title, num_cells, palette, ...)
+  z_out <- seq(min_depth, max_depth,by = interval) # Set plotting interval
+  # Get data from .nc file
+  data = get_var(file, var_name = var_name, z_out = z_out, reference = reference)
+  # Get units
+  units = sim_var_units(file, var_name = var_name)
+  
+  if (reference == 'surface'){
+    names.df = data.frame(names = names(data)[-1], depth.numeric = z_out, stringsAsFactors = F)
+    # ylabel = 'Depth (m)'
+  }
+  if (reference == 'bottom'){
+    names.df = data.frame(names = names(data)[-1], depth.numeric = rev(z_out), stringsAsFactors = F)
+    # ylabel = 'Elevation (m)'
+  }
+  dataLong = gather(data = data,key = depth, value = var, -DateTime) %>%
+    left_join(names.df, by = c('depth' = 'names')) 
+  names(dataLong)[3] = var_name
+  
+  if(is.null(legend.title)) {
+    legend.title = .unit_label(file, var_name)
+  }
+  .plot_df_heatmap(dataLong, var_name, legend.title, text.size, show.legend, legend.position, plot.title,
+                   color.palette, color.direction)
 }
 
-#' @importFrom graphics .filled.contour
-#' @importFrom grDevices colorRampPalette 
-#' @importFrom utils head
-.plot_df_heatmap <- function(data, bar_title, num_cells, palette, title_prefix=NULL, overlays=NULL, xaxis=NULL, col_lim){
-  
-  z_out <- rLakeAnalyzer::get.offsets(data)
-  reference = ifelse(substr(names(data)[2],1,3) == 'elv', 'bottom', 'surface')
-  
-  if (missing(col_lim))
-    col_lim = range(data[, -1], na.rm = TRUE)
-  if (missing(palette))
-    palette <- colorRampPalette(c("violet","blue","cyan", "green3", "yellow", "orange", "red"), 
-                                bias = 1, space = "rgb")
-  
-  col_subs <- head(pretty(col_lim, 6), -1)
-  levels <- sort(unique(c(col_subs, pretty(col_lim, 15))))
-  colors <- palette(n = length(levels)-1)
-  dates <- data[, 1]
-  matrix_var <- data.matrix(data[, -1])
-  if(is.null(xaxis)){
-  	xaxis <- get_xaxis(dates)
-  }
+
+.plot_df_heatmap <- function(dataLong, var_name, legend.title, text.size, 
+                             show.legend, legend.position, plot.title,
+                             color.palette, color.direction, ylabel) {
+                             
+                             # num_cells, palette, title_prefix=NULL, overlays=NULL, xaxis=NULL, col_lim){
+  
   
-  yaxis <- get_yaxis_2D(z_out, reference, prefix=title_prefix)
-  plot_layout(xaxis, yaxis, add=TRUE)
-  .filled.contour(x = dates, y = z_out, z =matrix_var,
-                  levels= levels,
-                  col=colors)
-  overlays # will plot any overlay functions
-  axis_layout(xaxis, yaxis) #doing this after heatmap so the axis are on top
+  h1 = ggplot(data = dataLong, aes(DateTime, depth.numeric)) +
+  geom_raster(aes_string(fill = var_name), interpolate = F, hjust = 0.5, vjust = 0.5, show.legend = show.legend) +
```

Comment:
I don't have much ggplot-fu, but shouldn't this all be indented after the `h1` line?